### ISSUE TITLE
diffUpdate producing incorrect results (example included)

### ISSUE THREAD

**Issue by AdventureBeard** [yjs/yjs#308]
I'm not sure if I'm doing something wrong or if I've found a bug, but I'm getting some strange behavior using differential updates that I don't see using the Y.Doc methods. This is causing us some trouble, resulting in divergence, etc.

**Describe the bug**

```javascript
let updateA = new Uint8Array([
  1, 1, 252, 169, 199, 206, 4,
  7, 40, 1, 4, 116, 101, 115,
  116, 3, 104, 101, 121, 1, 125,
  1, 0
])
let updateB = new Uint8Array([1, 4, 252, 169, 199, 206, 4, 0, 7, 1, 11, 112, 114, 111, 115, 101, 109, 105, 114, 114, 111, 114, 3, 9, 112, 97, 114, 97, 103, 114, 97, 112, 104, 7, 0, 252, 169, 199, 206, 4, 0, 6, 4, 0, 252, 169, 199, 206, 4, 1, 5, 97, 112, 112, 108, 101, 40, 1, 4, 116, 101, 115, 116, 3, 104, 101, 121, 1, 125, 1, 0])

let ydocA = new Y.Doc()
let ydocB = new Y.Doc()
Y.applyUpdate(ydocA, updateA)
Y.applyUpdate(ydocB, updateB)
let ydocA_sv = Y.encodeStateVector(ydocA)
let diffedUpdate1 = Y.encodeStateAsUpdate(ydocB, ydocA_sv) // this looks like the whole of updateB, and can be used to converge the docs
//Uint8Array(71) [1, 4, 252, 169, 199, 206, 4, 0, 7, 1, 11, 112, 114, 111, 115, 101, 109, 105, 114, 114, 111, 114, 3, 9, 112, 97, 114, 97, 103, 114, 97, 112, 104, 7, 0, 252, 169, 199, 206, 4, 0, 6, 4, 0, 252, 169, 199, 206, 4, 1, 5, 97, 112, 112, 108, 101, 40, 1, 4, 116, 101, 115, 116, 3, 104, 101, 121, 1, 125, 1, 0]

let sv = Y.encodeStateVectorFromUpdate(updateA)
let diffedUpdate2 = Y.diffUpdate(updateB, sv)
console.log(diffedUpdate2) // this update is empty
// Uint8Array [0,0]

```

**Expected behavior**
I'd expect to get a diff that could converge two docs based on updateA and updateB, but the diff is empty. 

**Environment Information**
 - Yjs 13.5.9
 - Chrome Version 90.0.4430.212 (Official Build) (x86_64) 
 - Node 14.5.0

**Comment by dmonad** [comment]
Hey @AdventureBeard, 

Could you please check again if the behavior is now correct?

It seems that the implementation of `encodeStateVectorFromUpdate` doesn't account for "missing updates". So you never requested missing updates that are necessary to actually sync. I'm sure it took you many hours to find this issue. Thanks for the help to detect it!

I just published Yjs@13.5.10 with a fix.

**Comment by AdventureBeard** [comment]
Yes, I don't see this issue anymore in development. We have been seeing strange behavior related to missing updates for weeks now; I wonder if this is the root cause? If so, it's a huge relief!  😄 I appreciate the quick fix.

Serendipitously, it was a race condition in our app that helped find this. It consistently caused update N+3 to be sent before the SS1 exchange between the client and server. The server would always indicate that it had update 1 & 2 and then the client would never send them.

I will deploy this to production ASAP and report back if anything novel occurs.
